src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -901,6 +914,12 @@ class kernel_command : public xrt_core::command
     return m_execbuf.first;
   }
 
+  xrt::hw_context
+  get_hw_context() const override
```

Comment:
I think it would be cheaper if we use `get_hwctx_handle() const`.  This will avoid reference counting magic, which while cheap is not for free.  But absolutely keep xrt::hw_context as the data member.

Reply:
fixed.